consumer/nf_mangement.go

```diff
@@ -194,42 +232,55 @@ func SendCreateSubscription(nrfUri string, nrfSubscriptionData models.NrfSubscri
 		}()
 		if res.Status != err.Error() {
 			logger.ConsumerLog.Errorf("SendCreateSubscription received error response: %v", res.Status)
-			return
+			return models.NrfSubscriptionData{}, nil, err
 		}
 		problem := err.(openapi.GenericOpenAPIError).Model().(models.ProblemDetails)
 		problemDetails = &problem
 	} else {
 		err = openapi.ReportError("server no response")
 	}
-	return
+	return models.NrfSubscriptionData{}, problemDetails, err
 }
 
-func SendRemoveSubscription(subscriptionId string) (problemDetails *models.ProblemDetails, err error) {
+func SendRemoveSubscription(ctx context.Context, subscriptionId string) (*models.ProblemDetails, error) {
 	logger.ConsumerLog.Infoln("[AMF] Send Remove Subscription")
 
+	ctx, span := tracer.Start(ctx, "HTTP DELETE nrf/subscriptions/{subscriptionID}")
+	defer span.End()
+
+	span.SetAttributes(
+		attribute.String("http.method", "DELETE"),
+		attribute.String("nf.target", "nrf"),
+		attribute.String("net.peer.name", amf_context.AMF_Self().NrfUri),
+		attribute.String("amf.nf.id", amf_context.AMF_Self().NfId),
+	)
+
 	amfSelf := amf_context.AMF_Self()
 	// Set client and set url
 	configuration := Nnrf_NFManagement.NewConfiguration()
 	configuration.SetBasePath(amfSelf.NrfUri)
 	client := Nnrf_NFManagement.NewAPIClient(configuration)
 	var res *http.Response
 
-	res, err = client.SubscriptionIDDocumentApi.RemoveSubscription(context.Background(), subscriptionId)
+	var problemDetails *models.ProblemDetails
+	var err error
+
+	res, err = client.SubscriptionIDDocumentApi.RemoveSubscription(ctx, subscriptionId)
```

Comment:
Same as my previous comments
```suggestion
func SendRemoveSubscription(ctx context.Context, subscriptionId string) (problemDetails *models.ProblemDetails, err error) {
	logger.ConsumerLog.Infoln("[AMF] Send Remove Subscription")

	ctx, span := tracer.Start(ctx, "HTTP DELETE nrf/subscriptions/{subscriptionID}")
	defer span.End()

	span.SetAttributes(
		attribute.String("http.method", "DELETE"),
		attribute.String("nf.target", "nrf"),
		attribute.String("net.peer.name", amf_context.AMF_Self().NrfUri),
		attribute.String("amf.nf.id", amf_context.AMF_Self().NfId),
	)

	amfSelf := amf_context.AMF_Self()
	// Set client and set url
	configuration := Nnrf_NFManagement.NewConfiguration()
	configuration.SetBasePath(amfSelf.NrfUri)
	client := Nnrf_NFManagement.NewAPIClient(configuration)
	var res *http.Response

	res, err = client.SubscriptionIDDocumentApi.RemoveSubscription(ctx, subscriptionId)
```

Reply:
Done